random-menu-generator.rb

```diff
@@ -0,0 +1,784 @@
+# def check_help_request(user_input)
+#   help = %{help what how when why where who}
+#   help_triggers.each do |help_trigger|
+#     if user_input.include? help_trigger
+#       return help_user(user_input)
+#     end
+#   end
+# end
+#
+#------------------------------ END HELP THINGS --------------------------------
+#----------------- BEGIN VERIFICATION / CONFIRMATION THINGS --------------------
+
+# gets input from a user, passes the input to check_reset_voice, and then
+# returns the user input string back to whatever requested it. because this is
+# the only way user input is harvested, any time the user says something it is
+# possible to toggle the voice on/off or trigger the currently not-so-helpful
+# help_user function (!W). has an option to trigger .downcase! on the user input
+# string before returning it.
+def get_user_input(downcase)
+  user_input = $stdin.gets.chomp
+
+  if downcase
+    user_input.downcase!
+  end
+
+  check_reset_voice(user_input)
+  # check_help_request(user_input) #!W !T
+
+  return user_input
+end
+
+
+
+
+# for use after yes/no questions. calls get_user_input, checks for key letters
+# in user response, and returns true/false accordingly. returns a random boolean
+# response if no key letters found.
+def request_user_confirmation
+  user_confirmation = get_user_input(true)
+
+  if user_confirmation.include? "n"
+    return false
+  elsif user_confirmation.include? "y"
+    return true
+  else
+    possible_responses = [true, false]
+    random_choice = rand(0...possible_responses.length)
+    return possible_responses[random_choice]
+  end
+end
+
+
+
+
+# checks if a user input string contains any keywords associated with various
+# features and then calls the next step for whichever feature.
+def verify_feeling(user_feeling)
+  creative = %w{creat add update more mine my own}
+  hungry = %w{hung angry peckish eat food dinner lunch breakfast dessert snack dish}
+  ingredients = %w{gred cook reci assem proc instr}
+  world_domination = %w{lazy bored take world hotdog elephant tacos}
+
+  creative.each do |keyword| # create is first, because 'eat' is part of it
+    if user_feeling.include? keyword
+      return create_user_ideas
+    end
+  end
+
+  hungry.each do |keyword| # eat is a trigger here, but create has precedence
+    if user_feeling.include? keyword
+      $individual = false
+      return retrieve_ideas("dishes")
+    end
+  end
+
+  ingredients.each do |keyword|
+    if user_feeling.include? keyword
+      $individual = true
+      return retrieve_ideas("hungry")
+    end
+  end
+
+  world_domination.each do |keyword|
+    if user_feeling.include? keyword
+      $individual = true
+      return retrieve_ideas("bored")
+    end
+  end
+
+
+  speak("You said #{user_feeling}. I'm sorry. I don't understand that emotion yet.")
+  speak("Until I learn your emotion, I will interpret that as hungry / angry.") # !W? this is sort of a lie b/c keyboard mismatch w/ name
+  $individual = true
```

Comment:
It feels a little confusing to call the catch-all "hungry / angry" but then set `$individual` to true here.